debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -96,103 +97,45 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
-                    if (message.isTransactionalMessage()) {
-                        if (!connectorConfig.shouldProvideTransactionMetadata()) {
-                            LOGGER.trace("Received transactional message {}", message);
-                            // Don't skip on BEGIN message as it would flush LSN for the whole transaction
-                            // too early
-                            if (message.getOperation() == Operation.COMMIT) {
-                                skipMessage(lsn);
-                            }
-                            return;
-                        }
-
-                        offsetContext.updateWalPosition(lsn, lastCompletelyProcessedLsn, message.getCommitTime(), message.getTransactionId(), null,
-                                taskContext.getSlotXmin(connection));
-                        if (message.getOperation() == Operation.BEGIN) {
-                            dispatcher.dispatchTransactionStartedEvent(Long.toString(message.getTransactionId()), offsetContext);
-                        }
-                        else if (message.getOperation() == Operation.COMMIT) {
-                            dispatcher.dispatchTransactionCommittedEvent(offsetContext);
-                        }
-                        maybeWarnAboutGrowingWalBacklog(true);
-                        return;
-                    }
-                    // DML event
-                    else {
-                        TableId tableId = null;
-                        if (message.getOperation() != Operation.NOOP) {
-                            tableId = PostgresSchema.parse(message.getTable());
-                            Objects.requireNonNull(tableId);
-                        }
-
-                        offsetContext.updateWalPosition(lsn, lastCompletelyProcessedLsn, message.getCommitTime(), message.getTransactionId(), tableId,
-                                taskContext.getSlotXmin(connection));
-
-                        boolean dispatched = (message.getOperation() == Operation.NOOP) ? false
-                                : dispatcher.dispatchDataChangeEvent(
-                                        tableId,
-                                        new PostgresChangeRecordEmitter(
-                                                offsetContext,
-                                                clock,
-                                                connectorConfig,
-                                                schema,
-                                                connection,
-                                                message));
-
-                        maybeWarnAboutGrowingWalBacklog(dispatched);
-                    }
-                });
-
-                if (receivedMessage) {
-                    noMessageIterations = 0;
+            if (walPosition.searchingEnabled()) {
+                searchWalPosition(context, stream, walPosition);
+                stream.stopKeepAlive();
```

Comment:
Why does keepalive need to be disabled to commit the current transaction?  Asking because PostgresTaskContext#refreshSchema called on https://github.com/debezium/debezium/pull/1751/files#diff-dff0ff397d3f72ba1319e51a37ace1adL118 here performs a commit but does not disable keepalive.

Reply:
Hi, in this case it is not due to the commit but due to the `reconnect` that is few lines later down